include/seqan3/alignment/pairwise/edit_distance_unbanded.hpp

```diff
@@ -416,44 +465,39 @@ struct pairwise_alignment_edit_distance_unbanded
         return res;
     }
 
-    //!\brief Compute the alignment.
-    decltype(auto) run()
-    {
-        // limit search width for prefix search
-        if constexpr(use_max_errors && is_global)
-        {
-            std::size_t max_length = query.size() + max_errors + 1;
-            std::size_t haystack_length = std::min(database.size(), max_length);
-            database_it_end += database.size() - haystack_length;
-        }
-
-        // distinguish between the version for needles not longer than
-        // one machine word and the version for longer needles
-        if (vp.size() <= 1)
-            small_patterns();
-        else
-            large_patterns();
-
-        return *this;
-    }
-
     //!\brief Return the score of the alignment.
-    int score()
+    score_type score() const noexcept
     {
-        return _score;
+        return -_best_score;
     }
 
     //!\brief Return the score matrix of the alignment.
-    score_matrix_type score_matrix()
+    score_matrix_type score_matrix() const noexcept
     {
         return score_matrix_type{*this};
     }
 
     //!\brief Return the trace matrix of the alignment.
-    trace_matrix_type trace_matrix()
+    trace_matrix_type trace_matrix() const noexcept
     {
         return trace_matrix_type{*this};
     }
+
+    //!\brief Return the end position of the alignment
+    alignment_coordinate end_coordinate() const noexcept
```

Comment:
still can we also compute the begin coordinate?

Reply:
yeah I have something in the pipeline